public/app/ui/components/controls/ListControlsComponent.js

```diff
@@ -15,22 +15,63 @@
                 };
 
             return function (parameters) {
-                if (!parameters || !parameters.pager) {
-                    throw new Error('ListControlsComponent missing required parameter: `pager`.');
+                if (!parameters) {
+                    throw new Error('ListControlsComponent missing parameter map.');
+                }
+                if (!parameters.modeSwitcher) {
+                    throw new Error('ListControlsComponent missing required parameter: `modeSwitcher`.');
                 }
-                if (!parameters.resultsMode) {
-                    throw new Error('ListControlsComponent missing required parameter: `resultsMode`.');
+                if (!parameters.pager) {
+                    throw new Error('ListControlsComponent missing required parameter: `pager`.');
                 }
-                if (!parameters.availableResultsModes) {
-                    throw new Error('ListControlsComponent missing required parameter: `availableResultsModes`.');
+                if (!parameters.sorter) {
+                    throw new Error('ListControlsComponent missing required parameter: `sorter`.');
                 }
                 if (!parameters.searchUrlFor) {
                     throw new Error('ListControlsComponent missing required parameter: `searchUrlFor`.');
                 }
 
-                this.availableResultsModes = ko.pureComputed(function () {
+                this.availableSortFields = ko.pureComputed(function () {
+                    return _.flatten(_.map(
+                        parameters.sorter.availableSortFields(),
+                        function (field) {
+                            return [
+                                {
+                                    label: field.labelText + ' asc',
```

Comment:
`asc` and `desc` are not necessarily 'in common parlance', at least label them as `ascending` and `descending`. I know space is an issue, but this is the public frontend so user training isn't going to cover it.